### PR TITLE
242 configuration screen

### PR DESCRIPTION
**Pull request by CedricCORGE**
New Interface for the container creation screen.

image
image
image

### CODE REVIEW

File: Configurateur/front/lib/screens/container-creation/design_screen.dart [hunk 1]
```diff
@@ -94,7 +94,7 @@ class DesignScreenState extends State<DesignScreen> {
 
   @override
   void initState() {
-    checkToken();
+    //checkToken();
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/design_screen.dart:97]: Why keep a commented line ?

File: Configurateur/front/lib/screens/container-creation/design_screen.dart [hunk 2]
```diff
@@ -167,7 +167,7 @@ class DesignScreenState extends State<DesignScreen> {
       if (faceLoad != null) {
         faceIndex = faceLoad;
       } else {
-        lockerss.add(Locker('design personnalisé', 50));
+        lockerss.add(Locker('design perso', 50));
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/design_screen.dart:170]: I don't think this should happen, try more correct language

File: Configurateur/front/lib/screens/container-creation/design_screen.dart [hunk 3]
```diff
@@ -247,7 +247,7 @@ class DesignScreenState extends State<DesignScreen> {
 
   void removeDesign(int faceIndex) {
     for (int i = 0; i < lockerss.length; i++) {
-      if (lockerss[i].type == 'design personnalisé') {
+      if (lockerss[i].type == 'design perso') {
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/design_screen.dart:250]: Same here

File: Configurateur/front/test/container-creation/container_creation_test.dart [hunk 4]
```diff
@@ -118,120 +118,6 @@ void main() {
     await tester.tap(find.byKey(const Key('terminate')));
   });
 
-  testWidgets('Container Creation back view panel from front',
```
Comment (31Nathan) [inline Configurateur/front/test/container-creation/container_creation_test.dart:121]: Why deleting so much tests ?

File: Configurateur/front/test/container-creation/design_screen_test.dart [hunk 5]
```diff
@@ -261,7 +261,7 @@ void main() {
     obj.materials[0] = FSp3dMaterial.grey.deepCopy()
       ..strokeColor = const Color.fromARGB(255, 0, 0, 255);
     designScreenState.objs.add(obj);
-    designScreenState.lockerss.add(Locker("design personnalisé", 50));
+    designScreenState.lockerss.add(Locker("design perso", 50));
```
Comment (31Nathan) [inline Configurateur/front/test/container-creation/design_screen_test.dart:264]: Correct language again

File: Configurateur/front/test/container-creation/design_screen_test.dart [hunk 6]
```diff
@@ -281,7 +281,7 @@ void main() {
...
```
Comment (31Nathan) [inline Configurateur/front/test/container-creation/design_screen_test.dart:284]: Same here

File: Configurateur/front/test/container-creation/design_screen_test.dart [hunk 7]
```diff
@@ -301,7 +301,7 @@ void main() {
...
```
Comment (31Nathan) [inline Configurateur/front/test/container-creation/design_screen_test.dart:304]: Same here

File: Configurateur/front/test/container-creation/design_screen_test.dart [hunk 8]
```diff
@@ -321,7 +321,7 @@ void main() {
...
```
Comment (31Nathan) [inline Configurateur/front/test/container-creation/design_screen_test.dart:324]: Again

File: Configurateur/front/test/container-creation/design_screen_test.dart [hunk 9]
```diff
@@ -341,7 +341,7 @@ void main() {
...
```
Comment (31Nathan) [inline Configurateur/front/test/container-creation/design_screen_test.dart:344]: And fix them all

File: Configurateur/front/lib/screens/container-creation/container_creation.dart [hunk 10]
```diff
@@ -21,11 +23,12 @@ import 'package:simple_3d_renderer/simple_3d_renderer.dart';
 
 import '../../components/dialog/autofill_dialog.dart';
 
+// ignore: must_be_immutable
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/container_creation.dart:26]: Not enough precisions

File: Configurateur/front/lib/screens/container-creation/container_creation.dart [hunk 11]
```diff
@@ -117,7 +120,7 @@ class ContainerCreationState extends State<ContainerCreation> {
 
     if (decodedContainer != null) {
       for (int i = 0; i < decodedContainer.length; i++) {
-        lockers.add(Locker('design personnalisé', 50));
+        lockers.add(Locker('design perso', 50));
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/container_creation.dart:123]: Language

File: Configurateur/front/lib/screens/container-creation/container_creation.dart [hunk 12]
```diff
@@ -462,6 +465,93 @@ class ContainerCreationState extends State<ContainerCreation> {
...
+      objs[0].fragments[fragment].faces[0].materialIndex = 0;
+      objs[0].fragments[fragment].faces[1].materialIndex = 0;
+      objs[0].fragments[fragment].faces[2].materialIndex = 0;
+      objs[0].fragments[fragment].faces[3].materialIndex = 0;
+      objs[0].fragments[fragment].faces[4].materialIndex = 0;
+      objs[0].fragments[fragment].faces[5].materialIndex = 0;
+      fragment += increment;
+    }
+
+    if (unitTesting == false) {
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/container_creation.dart:518]: What "unitTesting" variable is for ?

File: Configurateur/front/lib/components/recap_panel.dart [hunk 13]
```diff
@@ -33,11 +41,115 @@ class RecapPanel extends StatelessWidget {
...
+        },
+      );
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
+    parseArticles();
     return Container(
         //width: 300,
```
Comment (31Nathan) [inline Configurateur/front/lib/components/recap_panel.dart:154]: Why keeping commented lines ?

File: Configurateur/front/lib/components/dialog/container_dialog.dart [hunk 14]
```diff
@@ -103,42 +223,90 @@ class ContainerDialogState extends State<ContainerDialog> {
...
+                        } else {
+                          setState(() {
+                            face = '';
+                          });
+                        }
+                      },
+                      child: Text('Derrière',
+                          style: TextStyle(color: getTextColor())),
+                    ),
+                    /*Checkbox(
```
Comment (31Nathan) [inline Configurateur/front/lib/components/dialog/container_dialog.dart:290]: Why keeping comments ?

**Review by 31Nathan — CHANGES_REQUESTED** [review]

File: Configurateur/front/lib/screens/container-creation/design_screen.dart (hunk 1 above)
Comment (CedricCORGE) [reply Configurateur/front/lib/screens/container-creation/design_screen.dart:97]: I need it for testing, it's will be changed with the PR of design screen UI

File: Configurateur/front/test/container-creation/container_creation_test.dart (hunk 4 above)
Comment (CedricCORGE) [reply Configurateur/front/test/container-creation/container_creation_test.dart:121]: Don't exist anymore on the screen with the new interface so I remove it

**Pull request merged**